src/accelerate/utils/modeling.py

```diff
@@ -203,88 +203,6 @@ def id_tensor_storage(tensor: torch.Tensor) -> Tuple[torch.device, int, int]:
     return tensor.device, storage_ptr, storage_size
 
 
-def shard_checkpoint(
```

Comment:
Won't this be a backward breaking?

Reply:
Well, we can definitely keep that but I don't think anyone is actually using this function

Reply:
Agreed I don't *think* anyone is, but just in case let's deprecate to 0.33.0!

Reply:
Added back !